src/OrchardCore/OrchardCore.Lucene.Core/QueryProviders/SimpleQueryStringQueryProvider.cs

```diff
@@ -17,7 +19,18 @@ public Query CreateQuery(ILuceneQueryService builder, LuceneQueryContext context
             var queryString = query["query"]?.Value<string>();
             var fields = query["fields"]?.Values<string>() ?? new string[0];
             var defaultOperator = query["default_operator"]?.Value<string>().ToLowerInvariant() ?? "or";
-            var weights = fields.ToDictionary(field => field, field => 1.0f);
+            var weight = 1.0f;
+            var weights = fields.ToDictionary(field => field, field => weight);
+
+            foreach(var field in fields)
+            {
+                if(field.Contains("^", StringComparison.Ordinal) && Single.TryParse(field.Split("^").Last(), out weight))
```

Comment:
For infos, just before going to sleep, by default it doesn't use any comparisonType 

        public bool Contains(string value)
        {
            if (value == null)
                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.value);

            return SpanHelpers.IndexOf(
                ref _firstChar,
                Length,
                ref value._firstChar,
                value.Length) >= 0;
        }

Here the best is to pass a character `field.Contains('^')`

        public bool Contains(char value) => SpanHelpers.Contains(ref _firstChar, value, Length);

Reply:
I don't think it was needed because a '^' char might be the same in every culture. But here, I can't assert this fact because I'm not omniscient 😉